helm/qontract-reconcile/templates/template.yaml

```diff
@@ -127,8 +127,102 @@ objects:
                 key: log_group_name
           command: ["/bin/sh", "-c"]
           args:
+          # generate /etc/vector/vector.yaml
+          # and generate /fluentd/etc/fluent.conf
           - |
-            # generate fluent.conf
+            ### APPSRE-8096: Vector Section Begins
+            mkdir -p /etc/vector
+            cat > /etc/vector/vector.yaml <<EOF
+            sources:
+              qr:
+                type: file
+                include:
+                  - /fluentd/log/integration.log
+                read_from: beginning
+                data_dir: /fluentd/log
+
+            transforms:
+              ### De-spaminate
+              despam_events:
+                type: filter
+                inputs:
+                  - qr
+                condition:
+                  type: datadog_search
+                  source: -"using gql endpoint" AND -"Certificate did not match expected hostname"
+
+              ### Craft a JSON document that has escaped double quotes and changes the event from the 
+              ### Vector-default `message` to `text`, which is require by Slack and Google Chat.
+              rename_message_webhooks:
+                type: remap
+                inputs:
+                  - despam_events
+                source: |-
+                  .message = join!(
+                  [
+                    "{\"text\":\"",
+                    format_timestamp!(
+                      now(),
+                      "%Y-%m-%d %H:%M:%S %z"
+                    ),
+                    " ",
+                    " [${POD_NAME}] ",
+                    replace!(
+                      del(.message),
+                      "\"",
+                      "\\\""),
+                    "\"}"
+                  ])
+
+            sinks:
+              ################
+              ### Cloudwatch log sink
+              ################
+              cloudwatch_logs:
+                type: aws_cloudwatch_logs
+                inputs:
+                  - rename_message_webhooks
+                group_name: "${LOG_GROUP_NAME}"
+                encoding:
+                  codec: text
+                stream_name: "{{ $integration.name }}"
+
+            {{- if $logs.slack }}
+              ################
+              ### Slack sink
+              ################
+              slack:
+                type: http
+                inputs:
+                  - rename_message_webhooks
+                compression: none
+                encoding:
+                  codec: text
+                  only_fields:
+                    - message
+                method: post
+                uri: ${SLACK_WEBHOOK_URL}
+            {{- end }}
+
+            {{- if $logs.googleChat }}
+              ################
```

Comment:
Would it be possible to add filtering for Google Chat messages so that only ERROR level messages get sent? See https://github.com/app-sre/qontract-reconcile/pull/4420/files#diff-1b43156729c75dded2e7c94742402e1f8b0de449917632dccbe032d231396bd0R289 for how we do it with Vector.

Reason for this is that GChat really sucks at displaying log messages in real-time and will crash your browser lol

Reply:
Additionally, have you considered buffering to disk? We do so currently with GChat and Fluentd since GChat has some pretty strict rate limits. I worry a bit about the default of buffering to memory with a 120Mi limit.

Reply:
I'll look at the Error filtering.

I did look briefly at buffering to disk in local testing and for some reason it didn't quite work out? Maybe that was simply due to the small scale of local testing. I can add it in and we can see how it goes.

Reply:
Appreciate you doing the error filtering. We can see how buffering works out in FedRamp and then I'll let you know if we need to revisit logging to disk vs memory.

Reply:
@cubismod Sounds like a plan. 👍